### PR TITLE
Fix stale view indexing for per-tile coarse-tiled scratch buffers

### LINKED ISSUE #3003: Spyre coarse tiling mis-lowers unsqueeze consumer of amax result, causing incorrect denominator
### What happened?

A FlashAttention-like coarse-tiling case produces incorrect results when an `amax(..., dim=-1)` result is consumed through `unsqueeze(-1)` inside the same coarse-tiled fused kernel.

The issue appears to be in the generated layout for the `block_max.unsqueeze(-1)` input to a later `sub`. The `amax` output itself appears correct, but the subsequent read uses a device layout with an extra dimension that changes the physical row stride.

**Minimal Repro**
Script:

```text
test-spyre-scripts/repro_coarse_unsqueeze_denominator.py
```

Core compiled function:

```python
def denominator(q, k, mask):
    with spyre_hint(tiles={"B": 1}):
        with spyre_hint(tiles={"H": 1}):
            with spyre_hint(tiles={"Lq": 2}):
                with spyre_hint(tiles={"Lk": 1}):
                    with spyre_hint(work_div={"H": 4, "Lq": 8, "Lk": 8}):
                        scores = torch.matmul(q, k.transpose(-1, -2)) + mask
                        block_max = torch.amax(scores, dim=-1)
                        exp_scores = torch.exp(scores - block_max.unsqueeze(-1))
                        return exp_scores.sum(dim=-1)
```

The mask is all zeros, so it should not affect the math. It is included because it preserves the problematic fused/coarse-tiled graph shape.

**How To Run**
```bash
rm -rf /tmp/spyre_clean_repro_cache /tmp/spyre_clean_repro_debug

TORCHINDUCTOR_CACHE_DIR=/tmp/spyre_clean_repro_cache \
TORCH_COMPILE_DEBUG=1 \
TORCH_COMPILE_DEBUG_DIR=/tmp/spyre_clean_repro_debug \
SPYRE_INDUCTOR_LOG=1 \
SPYRE_INDUCTOR_LOG_LEVEL=debug \
python test-spyre-scripts/repro_coarse_unsqueeze_denominator.py
```

**Observed Result**
Example output:

```text
mismatches=3114/4096 max_abs=653.0
worst_index=(0, 3, 382) expected=289.75 actual=943.0
```

The generated logs show coarse tiling is active:

```text
coarse_tile_groups: 1 group(s) formed
LoopSpec(count=2)
```

**What To Look For**
Inspect the generated wrapper:

```bash
rg -n "device_size=\\[1, 512, 4, 2, 64\\]|LoopSpec|op='max'|op='sub'" \
  /tmp/spyre_clean_repro_cache -g '*.py'
```

Suspicious generated layout:

```python
# amax writes block_max with the expected tile layout:
device_size=[1, 512, 4, 64]
device_coordinates=[0, c1, c0, 0]

# later sub reads block_max.unsqueeze(-1) with an extra physical dimension:
device_size=[1, 512, 4, 2, 64]
device_coordinates=[0, c1, c0, 0, 0]
```

That extra `2` changes the physical stride of `c1`, so the `sub` appears to read the wrong row of `block_max`.

**Expected Result**
The compiled Spyre result should match the CPU reference within tolerance. The `unsqueeze(-1)` consumer of the `amax` result should preserve a layout compatible with the original `amax` output, rather than introducing a physical dimension that changes row addressing.

### What did you expect to happen?

the results should match

### How can we reproduce it?

Use this reproducer script
[repro_coarse_unsqueeze_denominator.py](https://github.com/user-attachments/files/29573009/repro_coarse_unsqueeze_denominator.py)

### Any environmental details we need to know?

No

### Anything else we need to know?

_No response_

### Relevant log output

```Shell
mismatches=3114/4096 max_abs=653.0
worst_index=(0, 3, 382) expected=289.75 actual=943.0
```

### PR DESCRIPTION
**Pull request by cyang49**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [x] bug
- [ ] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

This PR fixes a correctness/layout issue when coarse tiling is applied along an output dimension and an inside-loop view, such as `unsqueeze`, reads from a retiled scratch buffer.

In the failing flash-attention-style repro, coarse tiling correctly shrinks sparse intermediates from the full `Lq` extent to the per-tile extent, e.g.

```text
[1, 4, 2048] -> [1, 4, 512]
```

However, a `ReinterpretView` created by `unsqueeze` can still emit an index using the original full-tensor stride:

```text
2048*c0 + c1
```

When that stale index is interpreted against the retiled per-tile buffer layout, `compute_coordinates()` maps the full-stride factor into an extra device dimension. For `Lq=2048` and `q_block_size=512`, the generated wrapper previously contained:

```python
device_size=[1, 512, 4, 4, 64]
```

The extra `4` is the number of `Lq` tiles, not a real tensor dimension. The same issue appears as an extra `2` for `Lq=1024`, `q_block_size=512`.

This PR repairs the load index in `SpyreKernel.load()` for per-tile scratch buffers: if the load index coefficient still matches a pre-coarse-tiling full stride, it is rewritten back to the current per-tile stride before device-coordinate conversion.

Example repairs observed in debug logs:

```text
2048*c0 + c1 -> 512*c0 + c1
1024*c0 + c1 -> 512*c0 + c1
```

After the fix, the generated wrapper uses the expected device sizes:

```python
device_size=[1, 512, 4, 64]
device_size=[1, 512, 32, 64]
```

instead of:

```python
device_size=[1, 512, 4, 4, 64]
device_size=[1, 512, 32, 2, 64]
```
Generated wrapper artifacts were inspected to confirm the bad extra tile-count dimension is gone.

#### Which issue(s) this PR is related to:

Fixes #3003 

#### Does this PR introduce a user-facing change?

No

### CODE REVIEW

**Comment by cyang49** [comment]
solution too narrow.. abandoning
